### PR TITLE
Title in get layers and get maps

### LINKED ISSUE #88: Publish title attribute in the general request for list of layers
It would be convenient to display title instead of name in the client apps like HSLayers as title contains original name of the layer in human readable form (including special characters). Right now the GET request for list of layers (/rest/<user>/layers) returns only names of layers. Adding title to the response JSON like this would solve it

```
[
  {
    "name": "uzivatelske_kresleni",
    "title": "Uživatelské kreslení",
    "url": "https://hub.lesprojekt.cz/rest/fzadrazil/layers/uzivatelske_kresleni",
    "uuid": "d50de2bd-8169-4ea5-836b-20a08ca486c5"
  }
]
```

### PR DESCRIPTION
**Pull request by index-git**
Closes #88

### CODE REVIEW

File: src/layman/common/util.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+def get_names_from_infos(infos):
+    return list(set(sorted([info for info in infos])))
```
Comment (jirik) [inline src/layman/common/util.py:2]: If you need sorted, then set() makes is unsorted, sou you need to call set() first, sorted() second.

Comment (jirik) [reply src/layman/common/util.py:2]: But do you need sorted? I think not.

File: src/layman/layer/geoserver/wfs.py [hunk 2]
```diff
@@ -113,7 +114,7 @@ def mem_value_from_string_value(string_value):
         return wfs_proxy
 
     def currently_changing():
-        layernames = input_file.get_layer_names(username)
+        layernames = input_file.get_layer_infos(username)
```
Comment (jirik) [inline src/layman/layer/geoserver/wfs.py:117]: The variable layernames actually holds infos

File: src/layman/layer/rest_layers.py [hunk 3]
```diff
@@ -25,15 +25,16 @@ def before_request():
...
-            'name': layername,
-            'url': url_for('rest_layer.get', layername=layername, username=username),
-            'uuid': uuid.get_layer_uuid(username, layername),
+            'name': info["name"],
+            'title': info["title"],
+            'url': url_for('rest_layer.get', layername=name, username=username),
+            'uuid': info["uuid"],
         }
-        for layername in layernames
+        for (name, info) in layer_infos.items()
```
Comment (jirik) [inline src/layman/layer/rest_layers.py:37]: I prefer the resulted list to be sorted by layer name as it was.

File: src/layman/map/rest_maps.py [hunk 4]
```diff
@@ -27,15 +27,15 @@ def before_request():
...
         {
-            'name': mapname,
-            'url': url_for('rest_map.get', mapname=mapname, username=username),
-            'uuid': uuid.get_map_uuid(username, mapname),
+            'name': name,
+            'url': url_for('rest_map.get', mapname=name, username=name),
+            'uuid': uuid.get_map_uuid(username, name),
         }
-        for mapname in mapnames
+        for (name, info) in mapinfos.items()
```
Comment (jirik) [inline src/layman/map/rest_maps.py:38]: I prefer the resulted list to be sorted by map name as it was.

File: src/layman/map/rest_maps_test.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+            assert publication_names == module["result_publications"],\
+                (module["name"], module["method_publications"].__module__, publication_names)
+
+        map_infos = util.get_map_infos(username)
+        assert map_infos == result_infos_all, map_infos
+
+    client_util.delete_map(username, mapname, client)
+
+
+def test_get_map(client):
```
Comment (jirik) [inline src/layman/map/rest_maps_test.py:67]: test_get_map_title?

File: src/layman/map/rest_maps_test.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    mapname = 'test_get_map_infos_layer'
+    maptitle = "Test get map infos - map title íářžý"
+
+    client_util.publish_map(username, mapname, client, maptitle)
+
+    with app.app_context():
+        # maps.GET
+        rv = client.get(url_for('rest_maps.get', username=username))
+        assert rv.status_code == 200, rv.json
+        assert rv.json[0]["name"] == mapname, rv.json
```
Comment (jirik) [inline src/layman/map/rest_maps_test.py:78]: assert also maptitle?

**Review by jirik — CHANGES_REQUESTED** [review]

File: src/layman/common/util.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+
+def merge_infos(infos):
+    result_infos = {}
+    # TODO maybe, those two cycles can be done at once
```
[the remark this answers is not in the document]
Comment (index-git) [reply src/layman/common/util.py:7]: Zajímavá funkčnost.

File: src/layman/common/util.py (hunk 1 above)
Comment (index-git) [reply src/layman/common/util.py:2]: The reason was, that then we can change sorted() or unsorted variant in one place, not many.

**Review by jirik — APPROVED** [review]

**Pull request merged**